### ISSUE TITLE
embeddings.model is always "all-MiniLM-L6-v2" - Model() hardcodes the identity, so a compatible swap of the ONNX model is stored under a false label

### ISSUE THREAD

**Issue by sorin-sds** [arhuman/mnemos#31]
## Environment

- mnemos v0.2.0 (commit `78beb545dcec4eae131486cd4dc0fa3d80638a9b`), built with `go build -tags embed`
- Same code on `main` as of today (verified line-by-line below)
- Go 1.27.0, windows/amd64
- Model actually loaded: `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  (root `tokenizer.json` + `onnx/model.onnx`), placed in the model directory mnemos loads from

## What the code does

The storage layer is deliberately model-aware:

- `internal/storage/migrations/0002_embeddings.sql`: `embeddings(chunk_id, model TEXT, dim INTEGER, vector BLOB)` with `CREATE INDEX idx_embeddings_model`
- `internal/storage/embeddings.go` (v0.2.0 L39-40): *"dim and model are stored alongside so a **later model change can be detected**"*
- `internal/search/hybrid.go`: vector scans filter `WHERE e.model = ?` using `v.emb.Model()` (L78); `dotBlob` skips dimension-drifted rows

But the embedder's identity is hardcoded, so the model-aware plumbing can never see a second model:

| file (v0.2.0 = main) | line | code |
|---|---|---|
| `internal/embed/embedder.go` | 21 | `const DefaultModel = "all-MiniLM-L6-v2"` |
| `internal/embed/embedder.go` | 24 | `const Dim = 384` |
| `internal/embed/onnx.go` | 95 | `func (e *onnxEmbedder) Dim() int { return Dim }` |
| `internal/embed/onnx.go` | 98 | `func (e *onnxEmbedder) Model() string { return DefaultModel }` — returns the constant **regardless of what is in `modelDir`** |
| `internal/cli/semantic.go` | 20 | `dir, err := embed.ModelDir(embed.DefaultModel)` — the model path is not configurable |
| `internal/cli/models.go` | 31 | `modelDownloads` registry contains exactly one entry |

`New(modelDir)` takes a directory, but `Model()` ignores it. Any model that happens to satisfy the
current backend's assumptions (384-dim output, mean-pooling, `input_ids`/`attention_mask`/`token_type_ids`)
can be dropped into `~/.mnemos/models/all-MiniLM-L6-v2/` and will work — while every vector it produces
is labeled `"all-MiniLM-L6-v2"`.

## Minimal reproduction

1. `mnemos init && mnemos ingest <one-small-file-dir> --collection demo`
2. Replace `~/.mnemos/models/all-MiniLM-L6-v2/{model.onnx,tokenizer.json}` with the
   `paraphrase-multilingual-MiniLM-L12-v2` export (byte-identical to the HF files;
   `sha256[:16] = 10f7a088420252b2`, vs. the original MiniLM `6fd5d72fe4589f18`;
   its `tokenizer.json` reports `model.type = "Unigram"`, vocab 250 002 — not MiniLM's WordPiece 30 522)
3. `mnemos reindex --embeddings`
4. `sqlite3 .mnemos/mnemos.db "SELECT model, dim, COUNT(*) FROM embeddings GROUP BY 1,2"`

Observed:

```text
reindexed 1 embedding vectors (model all-MiniLM-L6-v2)
('all-MiniLM-L6-v2', 384, 1)
```

The vector was produced by paraphrase-multilingual-MiniLM-L12-v2; the database says
all-MiniLM-L6-v2. Retrieval works correctly — only the provenance is false.

## Expected behavior

`Model()` reports the identity of the model actually loaded (e.g. derived from config, the
model directory name, or a model manifest), so `embeddings.model` is truthful and the
already-implemented model filtering can distinguish vectors from different models.

## Actual behavior

`Model()` always returns the compile-time constant; `embeddings.model` is unconditionally
`"all-MiniLM-L6-v2"`.

## Impact

1. **False provenance.** Every vector row misattributes its producing model whenever the
   active ONNX file is not literally all-MiniLM-L6-v2. The `WHERE e.model = ?` filter and the
   `idx_embeddings_model` index cannot discriminate between different compatible models when
   `Model()` returns the same hardcoded identity.
2. **Indistinguishable mixed embeddings.** Two different 384-dim models used over time produce
   rows with the same `model` and same `dim`; the `dotBlob` drift guard cannot help. A model
   swap *without* a full reindex silently compares new-model query vectors against
   old-model document vectors under one label — degraded ranking with no error.
3. **Reindex / debugging / migration confusion.** CLI output (`reindexed N vectors (model …)`),
   DB inspection, and any future migration tooling all report the wrong model, which is exactly
   the scenario this spike hit: we had to hash the ONNX file to prove which model was active.

## Two separable aspects

- **Bug:** `onnxEmbedder.Model()` reports a compile-time identity instead of the loaded
  model's identity, contradicting the documented intent in `storage/embeddings.go`
  ("so a later model change can be detected").
- **Feature/config gap:** model name, model directory, `Dim` (hardcoded at `embedder.go:24`)
  and pooling are not configurable (`semantic.go:20`, `models.go:31`, `embedder.go:21/24`).
  The hardcoded `Dim()` is a configurability limitation rather than a demonstrated defect —
  it only becomes observable with a non-384-dim model. This issue does **not** ask for a
  backend redesign or multi-model runtime — only that the identity be truthful.

## Proposed minimal fix

- Derive the persisted model identity from what is actually loaded: e.g. a `[embedding] model =
  "<name>"` config key (defaulting to `all-MiniLM-L6-v2`) used by `loadEmbedder()` for both the
  directory *and* the string returned by `Model()`; or, at minimum, `Model()` returns the model
  directory's name.
- Optionally persist a small manifest (`model.json` with name/dim/pooling) alongside
  `model.onnx` at `models install` time so identity travels with the weights.
- `Dim` could likewise come from the parsed ONNX output shape rather than a constant, but that
  is an enhancement; the truthful label is the core ask.

Unrelated to #30 (CRLF panic); filed separately on purpose.

---

## AI-assisted authorship disclosure

This report was technically investigated, reproduced and drafted with the assistance of
the AI model qwen3.8-flash (OpenCode model ID `opencode-go/qwen3.8-flash`) through
OpenCode, and subsequently reviewed and supervised with ChatGPT. I am not a Go/Python
developer and English is not my primary language. I am publishing this issue after
personally supervising the investigation; all commands, reproduction steps, observed
results and the workaround described above were executed and verified locally before
submission.

**Comment by arhuman** [comment]
Fixed on `main` in b352e6b.

`Model()` now derives its identity from the model directory that was actually loaded, and a
new `[embedding].model` config key selects the model (defaulting to `all-MiniLM-L6-v2`).
Verified end to end with a real `paraphrase-multilingual-MiniLM-L12-v2` install:
`embeddings.model` and the reindex output both report the model that ran.

Thanks for separating the bug from the config gap so cleanly, and for flagging `Dim` as an
enhancement rather than folding it in. That made the scope obvious.

One limitation worth recording, since it bounds impact item 2: `embeddings.chunk_id` is the
PRIMARY KEY, so a chunk can only ever hold one vector. The truthful label makes a
stale-model state *detectable*, but two models still cannot coexist per chunk. After
switching models without a reindex, old vectors are now correctly labeled and simply do not
match the `WHERE e.model = ?` filter, so semantic search returns fewer results instead of
silently comparing mismatched vectors. Better failure mode, but still not a visible error.
I will track surfacing that as a warning separately.
